### PR TITLE
Add desktop-jvm module written with tornado fx

### PR DESCRIPTION
**Pull request by tokou**
A quickly hacked demo of a desktop-jvm module written in kotlin and based on TornadoFX
screen shot 2018-02-01 at 23 27 01

### CODE REVIEW

**Comment by hal9002** [comment]
Can one of the admins review this PR?

**Comment by tasomaniac** [comment]
OMG! This is amazing. ❤️ Love the community. Will review soon.

File: game-of-life-multiplatform/desktop-jvm/build.gradle [hunk 1]
```diff
@@ -0,0 +1,14 @@
+apply plugin: 'kotlin'
+
+dependencies {
+    compile "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
+    compile project(":common-jvm")
```
Comment (rock3r) [inline game-of-life-multiplatform/desktop-jvm/build.gradle]: Nit: could you use single quotes where we don't need interpolation please? (here and below)

**Comment by xrigau** [comment]
@hal9002 ok to test

File: game-of-life-multiplatform/desktop-jvm/src/main/kotlin/com/novoda/gol/presentation/DesktopAppView.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import com.novoda.gol.presentation.board.BoardViewInput
+import com.novoda.gol.presentation.board.apply
+import javafx.beans.property.SimpleObjectProperty
+import javafx.scene.Parent
+import javafx.scene.control.Button
+import javafx.scene.control.ComboBox
+import javafx.scene.layout.BorderPane
+import tornadofx.*
+
+class DesktopAppView : View() , AppView {
```
Comment (rock3r) [inline game-of-life-multiplatform/desktop-jvm/src/main/kotlin/com/novoda/gol/presentation/DesktopAppView.kt]: Nit: extra whitespace after `View()`

File: game-of-life-multiplatform/desktop-jvm/src/main/kotlin/com/novoda/gol/presentation/DesktopBoardView.kt [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    override var onStartSimulationClicked: () -> Unit = {}
+    override var onStopSimulationClicked: () -> Unit = {}
+
+    override fun renderBoard(boardEntity: BoardEntity) {
+        for (y in 0 until boardEntity.getHeight()) {
+            for (x in 0 until boardEntity.getWidth()) {
+                val cellAtPosition = boardEntity.cellAtPosition(x, y)
+                val rect = rectangles[y to x]
+                runLater {
+                    rect?.fill = if (cellAtPosition.isAlive) Color.BLUE
```
Comment (rock3r) [inline game-of-life-multiplatform/desktop-jvm/src/main/kotlin/com/novoda/gol/presentation/DesktopBoardView.kt]: Nit: I'd rather have `val rect = rectangles[y to x]!!` as we _must_ have a rectangle if the coordinates are inside valid bounds. Failing fast is probably preferable.

**Comment by tobiasheine** [comment]
@tokou Could u change the `README` to add your client?

File: game-of-life-multiplatform/desktop-jvm/src/main/kotlin/com/novoda/gol/presentation/DesktopAppView.kt [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    private lateinit var boardView: DesktopBoardView
+    private lateinit var controlButton: Button
+    private lateinit var patternSelectBox: ComboBox<PatternEntity>
+    private var selectedPatternProperty = SimpleObjectProperty<PatternEntity>()
+
+    override var onControlButtonClicked: () -> Unit = {}
+    override var onPatternSelected: (pattern: PatternEntity) -> Unit = {}
+
+    private var presenter = BoardPresenter(width = 20, height = 20)
```
Comment (tasomaniac) [inline game-of-life-multiplatform/desktop-jvm/src/main/kotlin/com/novoda/gol/presentation/DesktopAppView.kt]: can this be a `val`?

**Pull request merged**